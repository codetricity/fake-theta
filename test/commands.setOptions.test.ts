import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchMissingParameterError,
} from './helper';

describe('POST /osc/commands/execute setOptions', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.setOptions',
        parameters: {
          options: [
            { _filter: 'hdr' },
            { captureMode: 'image' },
            { exposureDelay: 5 },
          ],
        },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.setOptions', parameters: { session: 'dummy' } })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/commands/execute setOptions', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.setOptions',
        parameters: {
          options: { _filter: 'hdr', exposureDelay: 5 },
        },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.setOptions', parameters: { session: 'dummy' } })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/commands/execute setOptions', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.setOptions',
        parameters: {
          options: { _filter: 'hdr', exposureDelay: 5 },
        },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.setOptions', parameters: { session: 'dummy' } })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('Unknown Type POST /osc/commands/execute _setPluginOrders', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.setOptions',
        parameters: {
          options: { _filter: 'hdr', exposureDelay: 5 },
        },
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
