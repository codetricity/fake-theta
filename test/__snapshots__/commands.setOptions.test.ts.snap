// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute setOptions should respond with results 1`] = `
{
  "name": "camera.setOptions",
  "state": "done",
}
`;

exports[`THETA X POST /osc/commands/execute setOptions should respond with results 1`] = `
{
  "name": "camera.setOptions",
  "state": "done",
}
`;

exports[`THETA Z1 POST /osc/commands/execute setOptions should respond with results 1`] = `
{
  "name": "camera.setOptions",
  "state": "done",
}
`;
